binder/src/test/java/io/grpc/binder/RobolectricBinderSecurityTest.java

```diff
@@ -148,19 +122,13 @@ private static MethodDescriptor<Empty, Empty> getMethodDescriptor() {
         .build();
   }
 
-  private static class SomeService extends LifecycleService {
-
-    private final IBinderReceiver binderReceiver = new IBinderReceiver();
-    private final ArrayBlockingQueue<SettableFuture<Status>> statusesToSet =
-        new ArrayBlockingQueue<>(128);
-    private Server server;
-    private final ScheduledExecutorService scheduledExecutorService =
-        new MainThreadScheduledExecutorService();
-
-    @Override
-    public void onCreate() {
-      super.onCreate();
+  private final IBinderReceiver binderReceiver = new IBinderReceiver();
+  private final ArrayBlockingQueue<SettableFuture<Status>> statusesToSet =
+      new ArrayBlockingQueue<>(128);
+  private Server server;
 
+  @Before
+  public void setupServer() {
```

Comment:
Any particular reason for keeping 2 separate setup methods? This feels fragile if their execution order is non-deterministic. If this one triggers first, `listerAddress` will be unset.

Merging them would also allow us to reduce the scopes of some objects like `listenAddress`: [go/tott/343](https://goto.google.com/tott/343). You're welcome to keep it as a separate helper method called from  the other one though.

Reply:
I was just trying to minimize the diff but you are right it's better to put all this stuff in a conventional order.